tests/main.test.ts

```diff
@@ -47,22 +47,66 @@ describe("src/main.ts", function() {
 
     const validAny = validModel(anyType);
 
-    it('should allow all types when no restriction', function(done) {
+    it('should allow all types for -any- schema when there are no restriction',
+    function(done) {
       const validTypes = [
         undefined,
         null,
         5,
         "string",
         true,
-        {first: 1, second: "two"},
-        // null,
-        // undefined
+        {first: 1, second: "two"}
       ];
 
-      mochaAssertPromiseResovles(
-        Promise.all(validTypes.map(validAny)),
+      mochaAssertPromiseResovles(Promise.all(validTypes.map(validAny)), done);
+    });
+
+    const anyWithBoolRestriction: anySchema = {
+      isAny: true,
+      nullAllowed: true,
+      undefinedAllowed: true,
+      restriction: (someBool: boolean) => {
+        if(someBool) {
+          return Promise.reject("error");
+        }
+      }
+    };
+
+    const validAnyWithBoolRestriction = validModel(anyWithBoolRestriction);
+
+    it('should fail against the restriction', function(done) {
```

Comment:
" I tried to mimic the other tests though" 

` it('should not allow numbers if expecting a string'`
`it('should allow null if nullAllowed is set'`
` it('should not allow an object with extra properties'`

These are from the tests.

Perhaps you're referring to:

`it('should pass the restriction',`

Which indeed is not the best test name, but the point of that test is just to check that restrictions work at all. In your case, you're not just checking restrictions in general, you're doing more specific things. The more specific the test name the better, that way when a test fails we know exactly _what_ is broken.